src/unity/python/turicreate/data_structures/sframe.py

```diff
@@ -3144,8 +3144,18 @@ def add_column(self, data, column_name="", inplace=False):
         [3 rows x 3 columns]
         """
         # Check type for pandas dataframe or SArray?
+        
+        from collections import Iterable
+        
         if not isinstance(data, SArray):
-            raise TypeError("Must give column as SArray")
+            if isinstance(data, Iterable):
+                data = SArray(data)
+            else:
+                if len(self.column_names()) == 0:
+                    data = SArray([data])
+                else:
+                    data = SArray([data for _ in xrange(self.shape[0])])
```

Comment:
`xrange` has been removed as a keyword in Python 3. So this code will not work in Python 3. Writing code which works in both Python 2 and Python 3 can be a pain.

I would recommend just doing: `data = SArray([data] * self.shape[0])`